clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3576,14 +3576,25 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
 
   if (UseSYCLTriple) {
     // We want to compile sycl kernels.
-    if (types::isCXX(Input.getType()))
-      CmdArgs.push_back("-std=c++11");
     CmdArgs.push_back("-fsycl-is-device");
     // Pass the triple of host when doing SYCL
-    std::string NormalizedTriple =
-        llvm::Triple(llvm::sys::getProcessTriple()).normalize();
+    auto AuxT = llvm::Triple(llvm::sys::getProcessTriple());
+    std::string NormalizedTriple = AuxT.normalize();
     CmdArgs.push_back("-aux-triple");
     CmdArgs.push_back(Args.MakeArgString(NormalizedTriple));
+
+    bool IsMSVC = AuxT.isWindowsMSVCEnvironment();
+    if (types::isCXX(Input.getType()))
+      CmdArgs.push_back(IsMSVC ? "-std=c++14" : "-std=c++11");
```

Comment:
Right now we are having discussion with Alexey Bader on this. I say that even such a trivial test cannot be compiled with -std=c++11 due to the fact the MSVC headers use C++14 features:
#include <iostream>
int main() {  std::cout << "hello\n"; return 0; }

Alexey asks to unset _MSVC_LANG macro and see if we can use c++11 instead of c++14.

Reply:
why do we need to support c++11 when the headers are written for c++14 or 17?  how about we derive the right setting for c++ from the msvc version info?

Reply:
I had some experiments with that and still cannot make clang+MSVC working with -std=c++11, thus suggest continue with C++14 for now.